app/backend/core/messagebuilder.py

```diff
@@ -0,0 +1,31 @@
+from core.modelhelper import num_tokens_from_messages
+
+class MessageBuilder:
+  """
+    A class for building and managing messages in a chat conversation.
+    Attributes:
+        message (list): A list of dictionaries representing chat messages.
+        model (str): The name of the ChatGPT model.
+        token_count (int): The total number of tokens in the conversation.
+    Methods:
+        __init__(self, system_content: str, chatgpt_model: str): Initializes the MessageBuilder instance.
+        append_message(self, role: str, content: str, index: int = 1): Appends a new message to the conversation.
+        token_length(self) -> int: Returns the total token count in the conversation.
+        to_messages(self) -> list: Returns the list of messages in the conversation.
+    """
+  def __init__(self, system_content: str, chatgpt_model: str):
+    self.message = [{'role' : 'system', 'content': system_content}]
+    self.model = chatgpt_model
+    self.token_count = num_tokens_from_messages(self.message[-1], self.model)
+
+  def append_message(self, role:str, content: str, index: int = 1):
+    self.message.insert(index, {'role' : role, 'content': content})
+    self.token_count += num_tokens_from_messages(self.message[index], self.model)
+    return
+
+  def token_length(self) -> int:
```

Comment:
Why not just access token_count directly? Why the wrapper method?

Reply:
Comes from my previous experience with strong type languages. removed them :)

Reply:
No worries, that's what I figured. Python generally doesn't use getters/setters, since attributes are public by default. If you really want to protect an attribute, you can put a double underscore in front, and Python will make it harder (but not impossible) to access from outside the object. You can then use @property to make a method that looks like an attribute to access a dynamically calculated property.

In this case, I think it's fine to just have them be publicly accessible attributes (no underscores).